src/stringProcessing/specialCharacterMappings.js

```diff
@@ -0,0 +1,176 @@
+const includes = require( "lodash/includes" );
+const memoize = require( "lodash/memoize" );
+
+/**
+ * Gets indices of a specific character in the input text.
+ *
+ * @param {string} text The original text, for which the indices of specific characters have to be determined.
+ * @param {string} characterToFind The character that needs to be found in the text.
+ *
+ * @returns {Array} indices The array of indices in the text at which the characterToFind occurs.
+ */
+function getIndicesOfCharacter( text, characterToFind ) {
+	let indices = [];
+
+	if ( text.indexOf( characterToFind ) > -1 ) {
+		for( let i = 0; i < text.length; i++ ) {
+			if ( text[ i ] === characterToFind ) {
+				indices.push( i );
+			}
+		}
+	}
+
+	return indices;
+}
+
+/**
+ * Compares two arrays of which the second array is the sub-array of the first array.
+ * Returns the array of elements of the first array which are not in the second array.
+ *
+ * @param {Array} bigArray The array with all elements.
+ * @param {Array} subarray The array with some elements from the bigArray.
+ *
+ * @returns {Array} difference An array of all elements of bigArray which are not in subarray.
+ */
+function arraysDifference( bigArray, subarray ) {
+	return filter( bigArray, function( element ) {
+		return ( ! includes( subarray, element ) );
+	} );
+}
+
+
+/**
+ * Generates all possible combinations of the elements of an array (treated as unique).
+ * https://gist.github.com/jpillora/4435759
+ *
+ * @param {Array} array The array with elements that should be combined.
+ *
+ * @returns {Array} result An array of all possible combinations of elements of the original array.
+ */
+function combinations( array ) {
+	/**
+	 * A recursive function that iterates through all elements of an array to produce its combinations.
+	 *
+	 * @param {Array} xs The array to start accumulating with.
+	 * @param {Array} array The array with elements that should be combined.
+	 *
+	 * @returns {Array} result An array of all possible combinations of elements of the original array.
+	 */
+	function acc( xs, array ) {
+		const x = xs[ 0 ];
+
+		if( typeof x === "undefined" ) {
+			return array;
+		}
+
+		for( let i = 0, l = array.length; i < l; ++i ) {
+			array.push( array[ i ].concat( x ) );
+		}
+		return acc( xs.slice( 1 ), array );
+	}
+	return acc( array, [ [ ] ] ).slice( 1 ).concat( [ [] ] );
+}
+
+
+/**
+ * Replaces characters on specified indices in the input text.
+ *
+ * @param {string} text The original text, for which the characters have to be substituted.
+ * @param {Array} indices The array of indices that have to be substituted.
+ * @param {string} substitute The character that is used to substitute in the text.
+ *
+ * @returns {string} result The string of the original text with the characters on specified indices are substituted with the substitute character.
+ */
+function replaceCharactersByIndex( text, indices, substitute ) {
+	let modifiedTextSplitByLetter = text.split( "" );
```

Comment:
You can use `const` here because splice mutates the original string, it does not re-assign.